src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -33,14 +35,17 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
         self.cmd_clean_cache = "sudo tdnf clean expire-cache"
         self.cmd_repo_refresh = "sudo tdnf -q list updates"
 
+        # fetch snapshottime from health_store_id
+        self.snapshot_posix_time = self.__get_posix_time(execution_config.max_patch_publish_date, env_layer)
```

Comment:
self.env_layer is being set in the parent PackageManager

The pattern in the entire code base is to use that directly instead of passing it in as arguments within the scope of the class.

Reply:
Also, why is this being done in the constructor prior to configurepatching, assessment, etc.?

If the conversion fails, isn't it better to be specifically scoped to within the installpatches context?